plugins/convert_README_to_header.sh

```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
```

Comment:
Please can you stop making sweeping changes on chunks of this PR. The whole point of splitting them up was to be able to get small bits in at a time.

If you revert the sh/bash change, and all the changes to sed **except** the one to escape slashes or whatever the initial change was, then I believe this PR can be merged in it's current state. Yes it's not perfect, and there are known issues, but Travis still passes and it doesn't make things any worse. Then we can look at fixing the other bits in a separate PR. Likewise the base one is almost there apart from some trivial failures which Travis is already catching and flagging.